uyuni/common-libs/common/fileutils.py

```diff
@@ -502,13 +502,15 @@ def decompress_open(filename):
             import lzma
             file_obj = lzma.LZMAFile(filename, 'rb')
         except ImportError: # No LZMA lib - be sad
```

Comment:
Only in case if it's not included to `python-base`, I think it could happen if it was configured this way during package building, but not for openSUSE/SLE as it's there by default.

Reply:
Then I'd say such a fall-back mechanism is not needed. We only support openSUSE/SLE at this point and the requirements (in terms of packages) are well defined.

Reply:
uyuni-libs/common are also used on the clients, not only on the server. `lzma` in the standard library of Python 3, but not in Python 2.

If we want to keep Python2 compatibility, which I think we do, I would move the import statement to the top where it belongs and use one of the two common import guards:
```python
try:
    import lzma
except ImportError:
    lzma = None

if lzma is not None:
    file_obj = lzma.LZMAFile(...)
else:
    ...

# or

try:
    import lzma
    HAS_LZMA = True # could also be in an else clause
except ImportError:
    HAS_LZMA = False

if HAS_LZMA:
    file_obj = lzma.LZMAFile(...)
else:
    ...
```

Reply:
`decompress_open` is used by `reposync` and it's plugins only, I think it's quite safe if not to put import for this module to the global scope.

Reply:
> `decompress_open` is used by `reposync` and it's plugins only, 

I'm surprised this is in uyuni.common if it's only used on the server, afaik it was created for functionality common to server and clients.

> I think it's quite safe if not to put import for this module to the global scope.

It is safe in the sense that nothing else needs it and there won't be a traceback or anything like that, I agree. For compliance with [PEP 8](https://www.python.org/dev/peps/pep-0008/#imports) I would still move it to the top.

Reply:
In principle, anything which is not common should be moved out of the common library/package - but maybe in a different PR.

Reply:
I will discuss it with @mcalmer as he moved this part to common library, probably there are some reasons for it. I'll move the import to the top and handle the case if the import is failing

Reply:
fileutils are also used on clients. This special function (decompress_open) not (yet). In the past it was part of spacewalk-backend which required to build spacewalk-backend for all client OSes which was very hard as spacewalk-backend include a lot of other dependencies.
I just moved complete files over to uyuni-common and did not check every single call if it is used on clients or not.

I am fine with ignoring the python2 compatibility for this function or move this function back to spacewalk-backen.